examples/tv-casting-app/linux/main.cpp

```diff
@@ -237,23 +227,280 @@ void OnContentLauncherFailureResponse(void * context, CHIP_ERROR error)
+        }
+        for (auto & endpointInfo : mEndpoints)
+        {
+            if (endpointInfo.IsInitialized() && endpointInfo.GetEndpointId() == endpointId)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    void PrintInfo()
+    {
+        ChipLogProgress(NotSpecified, " TargetVideoPlayerInfo nodeId=0x" ChipLogFormatX64 " fabric index=%d",
+                        ChipLogValueX64(mNodeId), mFabricIndex);
+        for (auto & endpointInfo : mEndpoints)
+        {
+            if (endpointInfo.IsInitialized())
+            {
+                endpointInfo.PrintInfo();
+            }
+        }
+    }
+
+private:
+    static constexpr size_t kMaxNumberOfEndpoints = 5;
+    TargetEndpointInfo mEndpoints[kMaxNumberOfEndpoints];
+    NodeId mNodeId;
+    chip::FabricIndex mFabricIndex;
+    bool mInitialized = false;
+};
+TargetVideoPlayerInfo gTargetVideoPlayerInfo;
+
+void OnDescriptorReadSuccessResponse(void * context, const chip::app::DataModel::DecodableList<chip::ClusterId> & responseList)
+{
+    TargetEndpointInfo * endpointInfo = static_cast<TargetEndpointInfo *>(context);
+
+    ChipLogProgress(AppServer, "Descriptor: Default Success Response endpoint=%d", endpointInfo->GetEndpointId());
+
+    auto iter = responseList.begin();
+    while (iter.Next())
+    {
+        auto & clusterId = iter.GetValue();
+        endpointInfo->AddCluster(clusterId);
+    }
+    gTargetVideoPlayerInfo.PrintInfo();
+}
+
+void OnDescriptorReadFailureResponse(void * context, CHIP_ERROR error)
+{
+    ChipLogError(AppServer, "Descriptor: Default Failure Response: %" CHIP_ERROR_FORMAT, error.Format());
+}
+
+void ReadServerClusters(EndpointId endpointId)
+{
+    Server * server           = &(chip::Server::GetInstance());
+    chip::FabricInfo * fabric = server->GetFabricTable().FindFabricWithIndex(gTargetVideoPlayerInfo.GetFabricIndex());
+    if (fabric == nullptr)
+    {
+        ChipLogError(AppServer, "Did not find fabric for index %d", gTargetVideoPlayerInfo.GetFabricIndex());
+        return;
+    }
+
+    chip::DeviceProxyInitParams initParams = {
+        .sessionManager = &(server->GetSecureSessionManager()),
+        .exchangeMgr    = &(server->GetExchangeManager()),
+        .idAllocator    = &(server->GetSessionIDAllocator()),
+        .fabricTable    = &(server->GetFabricTable()),
+        .clientPool     = &gCASEClientPool,
+    };
+
+    PeerId peerID                                         = fabric->GetPeerIdForNode(gTargetVideoPlayerInfo.GetNodeId());
+    chip::OperationalDeviceProxy * operationalDeviceProxy = chip::Platform::New<chip::OperationalDeviceProxy>(initParams, peerID);
+    if (operationalDeviceProxy == nullptr)
+    {
+        ChipLogError(AppServer, "Failed in creating an instance of OperationalDeviceProxy");
+        return;
+    }
+
+    SessionHandle handle = server->GetSecureSessionManager().FindSecureSessionForNode(gTargetVideoPlayerInfo.GetNodeId());
```

Comment:
This will crash if we don't currently have an open CASE session to that node id.  What guarantees that we have such a CASE session?

Furthermore, this is not multi-fabric-safe...

As the comments on `FindSecureSessionForNode` say, we should not be using it.

Reply:
This was existing code. I'd prefer not to debug this in the current PR

Reply:
Again, had missed the existing callsite that was just being moved around.